--- a/Source/kwsys/SystemInformation.cxx
+++ b/Source/kwsys/SystemInformation.cxx
@@ -4936,0 +4937 @@
+  this->NumberOfLogicalCPU = info.cpu_count;
@@ -4964,0 +4966,2 @@
+  // Haiku exposes x86 CPUID fields only on x86 architectures.
+#if defined(__i386__) || defined(__x86_64__)
@@ -5003,0 +5007 @@
+#endif
--- a/Utilities/cmlibuv/CMakeLists.txt
+++ b/Utilities/cmlibuv/CMakeLists.txt
@@ -121,0 +122,10 @@
+
+if(CMAKE_SYSTEM_NAME STREQUAL "Haiku")
+  list(APPEND uv_libraries network)
+  list(APPEND uv_sources
+    src/unix/no-fsevents.c
+    src/unix/no-proctitle.c
+    src/unix/posix-hrtime.c
+    src/unix/posix-poll.c
+    )
+endif()
